notes/general/intro.md

```diff
@@ -0,0 +1,32 @@
+## Introduction
+
+### What is Algorithm?
+
+Informally, algorithms is well-defined sequence of steps that takes some values (input) and gives out some values after calculation (output).
+
+The words 'algorithm' and 'algorism' come from the name al-Khwārizmī. Al-Khwārizmī was a Persian mathematician, astronomer, geographer, and scholar. 
+
+### Example
+
+One of problem of sorting a set of elements in ascending or descending order can be defined formally as an algorithm as follows:
+
+_Input_ : A sequence of n numbers <a1,a2,...an>
+
+_Output_ : A permutation <a1', a2', ... an'> such that a1'<=a2'<=a3'...<=an'
+
+### Application of Algorithms
+
+* The Human Genome Project uses some advanced algorithms in identifying 100,000 genes in human DNA, determining the sequence of 3 billion chemical base pairs that make up human DNA and store this information in databases and retreive them and develop tools for analysis.
+
+* The Internet uses clever algorithms to allow websites to allow users to access large volumes of data without much delay.
+
+* E-Commerce websites utilize algorithms such as cryptography etc to allow users to have secure transactions over the Internet.
```

Comment:
"E-Commerce websites utilize algorithms such as cryptography etc to" > "E-Commerce websites utilize cryptographic algorithms to"